geopandas/geodataframe.py

```diff
@@ -684,10 +684,13 @@ def __setitem__(self, key, value):
         df['geometry'] = [geom... for geom in df.geometry]
         """
         if not pd.api.types.is_list_like(key) and key == self._geometry_column_name:
-            try:
-                value = _ensure_geometry(value, crs=self.crs)
-            except TypeError:
-                warnings.warn("Geometry column does not contain geometry.")
+            if value is None:
+                value = GeometryArray(np.full(self.shape[0], None), crs=self.crs)
```

Comment:
For scalars in general, we can probably fill an empty object array of the correct shape with that value, and then pass-through `_ensure_geometry`

Reply:
Good point with scalars. I have used simple `value = [value] * self.shape[0]`. Filling NumPy array with shapely geometry proved to be unnecessarily tricky as it is list like itself and NumPy tries to unpack it.